tests/static/js/formatters.js

```diff
@@ -153,4 +153,43 @@ describe('Formatters', () => {
       expect(actual).toEqual(expected);
     });
   });
+
+  describe('getCategoryNames', () => {
+    const categoryMap = [ 
+      {
+        "id": "1",
+        "category": "Neurology"
+      },
+      {
+        "id": "2",
+        "category": "Dermatology"
+      },
+      {
+        "id": "3",
+        "category": "Psychiatry"
+      },
+      {
+        "id": "4",
+        "category": "Surgery"
+      }
+    ];
+
+    it('Handle undefined categoryIds and categoryMap', () => {
+      let categoryNames = Formatters.getCategoryNames(null, categoryMap);
+      expect(categoryNames).toEqual([]);
+      categoryNames = Formatters.getCategoryNames(['1'], null);
+      expect(categoryNames).toEqual([]);
+    });
+
+    it('return empty list for no matching category names', () => {
+      const categoryIds = ['5', '0'];
+      const categoryNames = Formatters.getCategoryNames(categoryIds, categoryMap);
+      expect(categoryNames).toEqual([]);
+    });
+    it('return a list of matching category names', () => {
+      const categoryIds = ['1', '3'];
+      const categoryNames = Formatters.getCategoryNames(categoryIds, categoryMap);
+      expect(categoryNames).toEqual(['Neurology', 'Psychiatry']);
+    });
+  });
```

Comment:
do we want to add a test for a mix of matching and not matching category names?

Reply:
sure, updated!